src/main/msp/msp_serial.c

```diff
@@ -318,7 +318,7 @@ static int mspSerialSendFrame(mspPort_t *msp, const uint8_t * hdr, int hdrLen, c
     // Transmit frame
     serialBeginWrite(msp->port);
     serialWriteBuf(msp->port, hdr, hdrLen);
-    serialWriteBuf(msp->port, data, dataLen);
+    serialWriteBufNoFlush(msp->port, data, dataLen);
     serialWriteBuf(msp->port, crc, crcLen);
```

Comment:
What about hdr and crc? Both will call beginWrite/endWrite, so serialWriteBufNoFlush is called without begin and serialEndWrite is called twice.

It will probably work, but it is not worth the headache

Reply:
Not sure how I missed that. Of course hdr and crc should both be written using `serialWriteBufNoFlush()`.